File: tab.py
#!/usr/bin/python3
# Tabulate Borda Count votes

from sys import stdin

votes = { c:0 for c in 'abcdef-' }

for line in stdin:
    for i, c in enumerate(line.rstrip()):
        votes[c] += 6 - i

# http://stackoverflow.com/questions/613183
for k in sorted(votes, key=votes.get, reverse=True):
    if k != '-':
        print('{}: {}'.format(k, votes[k]))

File: tab.rs
// Tabulate Borda Count votes.

use std::io::prelude::*;
use std::io;

#[derive(Copy, Clone)]
struct Vote {
    candidate: char,
    tally: usize
}

fn main() {
    let mut votes = [Vote{candidate: '?', tally: 0}; 6];
    for i in 0..6 {
        votes[i].candidate = (i + ('a' as usize)) as u8 as char
    };
    let stdin = io::stdin();
    for maybe_line in stdin.lock().lines() {
        let line = maybe_line.expect("failed to read input line");
        assert!(line.len() == 6, "wrong input line size");
        let carray: Vec<char> = line.chars().collect();
        for i in 0..6 {
            let c = carray[i];
            if c == '-' {
                continue
            };
            assert!(c >= 'a' && c <= 'f', "bad vote char");
            votes[(c as usize - 'a' as usize)].tally += 6 - i
        }
    };
    votes.sort_unstable_by_key(|v| v.tally);
    for i in (0..6).rev() {
        println!("{}: {}", votes[i].candidate, votes[i].tally)
    }
}
